### PR TITLE
Allow the Current Ruleset to be changed

### PR DESCRIPTION
**Pull request by frank-weinberg**
This allows e.g. changing intermission duration or en/disabling auto start/stop rules without a reset of the teams.

### CODE REVIEW

File: src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java [hunk 1]
```diff
@@ -62,6 +63,15 @@ else if (flag != Flag.INTERNAL) {
         if (prop == Value.INVERTED_TIME) {
             return getMaximumTime() - getTime();
         }
+        if (prop == Value.MAXIMUM_TIME && getId() == ID_INTERMISSION) {
+            long duration = DEFAULT_MAXIMUM_TIME;
```
Comment (brian-brazil) [inline src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java]: This means I can no longer override the maximum time for an intermissions from the operator screen, which is often required when the schedule needs moving forward.

**Comment by brian-brazil** [comment]
That updating/deleting rulesets that are in use will affect the in-progress game may not be completely obvious to users.

**Comment by frank-weinberg** [comment]
I have removed the delete button (because it's completely unclear what should be done if the active ruleset is deleted) and added a note that editing will affect the current game for rulesets currently in effect.
(I've also changed handling of max intermission time so manual overrides are possible again.)

**Comment by brian-brazil** [comment]
That works.

A few small issues:
- Changing the current ruleset dosen't update the Delete button/warning
- If you create a ruleset that's only whitespace, things get odd
- The Name of "WFTDA Sanctioned" shouldn't be in a textbox, as it's not editable

I suspect the last two are existing.

**Comment by frank-weinberg** [comment]
They were existing but sufficiently easy fixes/workarounds.

File: html/components/rulesets-tab.js [hunk 2]
```diff
@@ -261,6 +302,9 @@ function createRulesetsTab(tab) {
 				}
 				WS.Set("ScoreBoard.Rulesets.Ruleset(" + activeRuleset.Id + ").Rule(" + val.Fullname + ")", value);
 			});
+			var newName = tab.find("#name").val();
+			if (newName.trim() === '') { newName = 'Unnamed'; }
```
Comment (brian-brazil) [inline html/components/rulesets-tab.js]: Elsewhere (colors) we ignore such requests, I think we should do the same here.

File: html/components/rulesets-tab.js [hunk 3]
```diff
@@ -225,10 +258,17 @@ function createRulesetsTab(tab) {
 		});
 
 		if (rs.Immutable) {
+			tab.find("#name").attr('readonly', true);
```
Comment (brian-brazil) [inline html/components/rulesets-tab.js]: I think disabled would be a bit clearer

**Review by brian-brazil — COMMENTED** [review]
Grand, saves me from fixing them.

**Comment by frank-weinberg** [comment]
Addressed the last comments and fixed an NPE discovered in the process.

**Pull request merged**